CI-API.Data/Repository/AdminRepository.cs

```diff
@@ -165,37 +185,114 @@ public async Task<JsonResult> getUser(long? userId)
         #region UpdateUserData
         public async Task<JsonResult> updateUserData(UserDetailViewModel userDetailViewModel)
         {
+            try
+            {
+
+                if (userDetailViewModel != null)
+                {
+                    if (userDetailViewModel.userId!=0 )
+                    {
+                        byte[] byteForPassword = Encoding.ASCII.GetBytes(userDetailViewModel.password);
```

Comment:
Make private functions for Update and Add user. and use that functions here. to make code readable.